### PR TITLE
Add nodejs example

### PR DESCRIPTION
**Pull request by xzd9326**
Added typescript examples based on the current build.

### CODE REVIEW

File: examples/nodejs/index.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import { Interface, JsonRpcProvider, Network } from "ethers";
```
Comment (novaliu86) [inline examples/nodejs/index.ts:1]: For the latest code change, probably we need dynamic imports here.

File: examples/nodejs/index.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            }
+        ],
+        "payable": false,
+        "stateMutability": "view",
+        "type": "function"
+    }]`;
+
+  const rpcUrl = "https://eth.meowrpc.com";   // Mainnet RPC URL
+
+  const network = { chainId: 1, name: "unknown" };
```
Comment (novaliu86) [inline examples/nodejs/index.ts:27]: `network` is not used.

File: examples/nodejs/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "extends": "@tsconfig/node18/tsconfig.json",
+  "compilerOptions": {
+    "target": "es2022",
+    "outDir": "dist",
+    "module": "nodenext",
+    "moduleResolution": "nodenext",
+    "resolveJsonModule": true
```
Comment (novaliu86) [inline examples/nodejs/tsconfig.json]: Probably we don't need this line anymore as no json modules are imported.

File: examples/nodejs/yarn.lock
Comment (novaliu86) [inline examples/nodejs/yarn.lock:1]: Do not provide a lock file for examples.

File: yarn.lock
Comment (novaliu86) [inline yarn.lock:1]: This change, if necessary, should be in a separate PR.

File: examples/nodejs/package.json
Comment (novaliu86) [inline examples/nodejs/package.json:1]: Consider changing this example project to esm project with `type: module` in package.json. Use `index.ts` file or `index.mts` file for esm use case, and also include an index.cjs file to demonstrate the commonjs use case.

File: examples/nodejs/index.ts (hunk 2 above)
Comment (xzd9326) [reply examples/nodejs/index.ts:27]: It is actually used on line 30.

File: examples/nodejs/yarn.lock
Comment (xzd9326) [reply examples/nodejs/yarn.lock:1]: Removed.

File: yarn.lock
Comment (xzd9326) [reply yarn.lock:1]: Will be included in another PR

**Review by novaliu86 — APPROVED** [review]
